# 2022 QB Data

In [4]:
import numpy as np
import pandas as pd
import re
import seaborn as sns
import matplotlib.pyplot as plt
import cv2
from IPython.display import Image

df = pd.read_csv('https://raw.githubusercontent.com/jonpb99/Fantasy_Football_Data_Analytics/main/2022_Data/ff_research_qb.csv')
df.head()

Unnamed: 0  Projected Points  Pre-Season Ranking    Wk1  Wk1_Total  \
0          Josh Allen               NaN                   1  33.48      33.48   
1       Lamar Jackson               NaN                   2  21.22      21.22   
2      Justin Herbert               NaN                   3  23.26      23.26   
3  Patrick Mahomes II               NaN                   4  34.90      34.90   
4        Kyler Murray               NaN                   5  20.62      20.62   

   Wk1_Rank    Wk2  Wk2_Total  Wk2_Rank    Wk3  ...   Wk13  Wk13_Total  \
0       NaN  29.68        NaN       NaN  26.70  ...  16.92         NaN   
1       NaN  42.62        NaN       NaN  40.42  ...   1.34         NaN   
2       NaN  24.46        NaN       NaN  13.58  ...  18.10         NaN   
3       NaN  17.30        NaN       NaN  18.08  ...  19.82         NaN   
4       NaN  26.88        NaN       NaN  13.36  ...    NaN         NaN   

   Wk13_Rank   Wk14  Pts Post Wk 14  Rank Post Wk 14   Wk15  Unnamed: 46  \
0        NaN  20.58             NaN              NaN  35.86          NaN   
1        NaN   0.00             NaN              NaN   0.00          NaN   
2        NaN  19.48             NaN              NaN  10.72          NaN   
3        NaN  22.78             NaN              NaN  32.74          NaN   
4        NaN   0.66             NaN              NaN   0.00          NaN   

   Unnamed: 47   Wk16  
0          NaN  24.98  
1          NaN   0.00  
2          NaN   6.50  
3          NaN  23.76  
4          NaN   0.00  

[5 rows x 49 columns]

In [2]:
df2 = pd.read_csv('https://raw.githubusercontent.com/jonpb99/Fantasy_Football_Data_Analytics/main/2022_Data/ff_research_qb2.csv')
df2.head()

NAME  RTG TEAM  DEP  BYE  GP  FPTS  SNAP  TOUCH    #  ...  ATT  \
0  Patrick Mahomes   97   KC  1.0    8   1  34.9   1.9    0.1   65  ...   39   
1       Josh Allen   95  BUF  1.0    7   1  33.5   1.8    0.3   59  ...   31   
2      Jalen Hurts   86  PHI  1.0    7   1  24.7   3.1    0.7   77  ...   32   
3   Justin Herbert   87  LAC  1.0    8   1  23.3   2.9    0.2   67  ...   34   
4       Joe Burrow   94  CIN  1.0   10   1  22.2   4.5    0.3  100  ...   53   

   COMP  YDS  TD  INT  300+  RUSH  RUSHYDS  RUSHTD  WEEK  
0    30  360   5    0     1     3        5       0     1  
1    26  297   3    2     0    10       56       1     1  
2    18  243   0    0     0    17       90       1     1  
3    26  279   3    0     0     4        1       0     1  
4    33  338   2    4     1     6       47       0     1  

[5 rows x 21 columns]

### Josh Allen

In [5]:
Image(url="ja.jpg")

In 2022, Josh Allen was the highest drafted QB. It made sense, since he was the QB1 (1st ranked QB in terms of fantasy points scored) for two seasons straight. Football analysts both inside and outside the fantasy world debated whether he might be the NFL's best quarterback over Patrick Mahomes. Since 2020, Josh Allen's passing stats were elite, but behind Mahomes. However, Allen was twice as likely to run the ball, and more effective than Mahomes when doing so. His running style was as exciting as it was unconventional, as he elected to run through defenders as opposed to around them, sometimes surprising people by jumping over them completely. Big and athletic, this becames a staple of his game. It all culminated in an exciting playoff game in the 2021 season against Mahomes' Chiefs where both quarterbacks put on a show. When Buffalo lost in overtime without Josh Allen even touching the ball, it caused the NFL to change the overtime rules for future playoffs.

This year, Josh Allen was the odds favorite to win MVP. The Bills were the favorite to win the superbowl. Buffalo made their team stronger by boosting their already great defense in free agency, and drafting a running back to help the offense. They also expected their promising young receiver (Gabriel Davis) to take the next step in his development. On the other hand, Patrick Mahomes lost his best receiver (Tyreek Hill) to Miami in the offseason. Operating an offense where the best pass catcher is an aging TE in his 9th year would be a challenge for any quarterback, but it was one Kansas City was comfortable with.

So how did Josh Allen perform in 2022? As expected? Better than expected? Worse then expected?

In short, he played well. As the QB2 for this season (QB3 on a per-game basis), I would say he performed as expected. Its possible you wanted the top-ranked quarterback in the fantasy draft to be an automatic QB1. However, whenever I draft a top QB (or TE, RB, WR), I expect stability. Things happen in any given season. Josh Allen was top 3 in total points and points per game. The QB1 in total points won MVP, and the QB1 in points per game finished second in MVP voting, just ahead of Josh Allen.

So while he did play well, it wasn't perfect. Let's see what worked against him, and if it is cause for concern for future seasons.

In [6]:
df_ja = df2[df2['TEAM']=='BUF']
df_ja = df_ja[['NAME','TEAM','FPTS','ATT','COMP','YDS','TD','INT','300+','RUSH','RUSHYDS','RUSHTD','WEEK']]
df_ja.describe()

FPTS        ATT       COMP        YDS         TD        INT  \
count  15.000000  15.000000  15.000000   15.00000  15.000000  15.000000   
mean   25.793333  35.733333  22.666667  268.60000   2.133333   0.866667   
std     6.823014   9.617444   7.187953   82.15821   1.245946   0.915475   
min    12.600000  25.000000  13.000000  147.00000   0.000000   0.000000   
25%    21.100000  29.000000  18.000000  209.00000   1.000000   0.000000   
50%    26.400000  34.000000  22.000000  253.00000   2.000000   1.000000   
75%    30.300000  40.000000  26.000000  323.00000   3.000000   2.000000   
max    36.200000  63.000000  42.000000  424.00000   4.000000   2.000000   

            300+       RUSH    RUSHYDS     RUSHTD      WEEK  
count  15.000000  15.000000  15.000000  15.000000  15.00000  
mean    0.400000   7.666667  49.733333   0.466667   8.60000  
std     0.507093   3.109126  25.686202   0.639940   4.91063  
min     0.000000   1.000000   7.000000   0.000000   1.00000  
25%     0.000000   6.000000  36.500000   0.000000   4.50000  
50%     0.000000   8.000000  47.000000   0.000000   9.00000  
75%     1.000000  10.000000  73.500000   1.000000  12.50000  
max     1.000000  12.000000  86.000000   2.000000  16.00000

In [7]:
df_ja

NAME TEAM  FPTS  ATT  COMP  YDS  TD  INT  300+  RUSH  RUSHYDS  \
1    Josh Allen  BUF  33.5   31    26  297   3    2     0    10       56   
13   Josh Allen  BUF  29.7   38    26  317   4    0     1     1       10   
23   Josh Allen  BUF  26.7   63    42  400   2    0     1     8       47   
32   Josh Allen  BUF  24.5   36    19  213   1    1     0    11       70   
39   Josh Allen  BUF  36.2   31    20  424   4    1     1     5       42   
49   Josh Allen  BUF  26.4   40    27  329   3    0     1    12       32   
68   Josh Allen  BUF  19.6   25    13  218   2    2     0     6       49   
73   Josh Allen  BUF  26.8   34    18  205   0    2     0     9       86   
82   Josh Allen  BUF  21.6   43    29  330   1    2     1     6       84   
94   Josh Allen  BUF  12.6   27    18  197   1    0     0     3        7   
96   Josh Allen  BUF  30.9   42    24  253   2    1     0    10       78   
110  Josh Allen  BUF  16.9   33    22  223   2    0     0     8       20   
116  Josh Allen  BUF  20.6   27    16  147   1    0     0    10       47   
122  Josh Allen  BUF  35.9   40    25  304   4    0     1    10       77   
129  Josh Allen  BUF  25.0   26    15  172   2    2     0     6       41   

     RUSHTD  WEEK  
1         1     1  
13        0     2  
23        0     3  
32        1     4  
39        0     5  
49        0     6  
68        0     8  
73        2     9  
82        0    10  
94        0    11  
96        1    12  
110       0    13  
116       1    14  
122       0    15  
129       1    16

These charts mainly paint Josh Allen in a favorable light. The second one shows his stats for each week. I tend to expect my QB (superstar or not) to score 20 fantasy points (FPTS) for me to consider it a good week. Allen crossed that threshold consistently, missing the mark only 3 times in 2022. He only had one 'bad' performance in week 11, scoring only 12.6 FPTS.

However, I do remember analysts pointing out Josh Allen not playing up to his standard halfway through the season. If you split the stats, you may find some interesting discrepancies. Josh Allen's stats look a bit different before and after Buffalo's week 7 bye week

In [8]:
df_ja_prebye = df_ja.iloc[[0,1,2,3,4,5]]
df_ja_prebye

NAME TEAM  FPTS  ATT  COMP  YDS  TD  INT  300+  RUSH  RUSHYDS  \
1   Josh Allen  BUF  33.5   31    26  297   3    2     0    10       56   
13  Josh Allen  BUF  29.7   38    26  317   4    0     1     1       10   
23  Josh Allen  BUF  26.7   63    42  400   2    0     1     8       47   
32  Josh Allen  BUF  24.5   36    19  213   1    1     0    11       70   
39  Josh Allen  BUF  36.2   31    20  424   4    1     1     5       42   
49  Josh Allen  BUF  26.4   40    27  329   3    0     1    12       32   

    RUSHTD  WEEK  
1        1     1  
13       0     2  
23       0     3  
32       1     4  
39       0     5  
49       0     6

In [9]:
df_ja_prebye.describe()

FPTS        ATT       COMP         YDS        TD       INT  \
count   6.000000   6.000000   6.000000    6.000000  6.000000  6.000000   
mean   29.500000  39.833333  26.666667  330.000000  2.833333  0.666667   
std     4.547087  11.923366   8.238123   75.741666  1.169045  0.816497   
min    24.500000  31.000000  19.000000  213.000000  1.000000  0.000000   
25%    26.475000  32.250000  21.500000  302.000000  2.250000  0.000000   
50%    28.200000  37.000000  26.000000  323.000000  3.000000  0.500000   
75%    32.550000  39.500000  26.750000  382.250000  3.750000  1.000000   
max    36.200000  63.000000  42.000000  424.000000  4.000000  2.000000   

           300+       RUSH    RUSHYDS    RUSHTD      WEEK  
count  6.000000   6.000000   6.000000  6.000000  6.000000  
mean   0.666667   7.833333  42.833333  0.333333  3.500000  
std    0.516398   4.167333  20.614720  0.516398  1.870829  
min    0.000000   1.000000  10.000000  0.000000  1.000000  
25%    0.250000   5.750000  34.500000  0.000000  2.250000  
50%    1.000000   9.000000  44.500000  0.000000  3.500000  
75%    1.000000  10.750000  53.750000  0.750000  4.750000  
max    1.000000  12.000000  70.000000  1.000000  6.000000

In [10]:
df_ja_postbye = df_ja.iloc[[6,7,8,9,10,11,12,13,14]]
df_ja_postbye

NAME TEAM  FPTS  ATT  COMP  YDS  TD  INT  300+  RUSH  RUSHYDS  \
68   Josh Allen  BUF  19.6   25    13  218   2    2     0     6       49   
73   Josh Allen  BUF  26.8   34    18  205   0    2     0     9       86   
82   Josh Allen  BUF  21.6   43    29  330   1    2     1     6       84   
94   Josh Allen  BUF  12.6   27    18  197   1    0     0     3        7   
96   Josh Allen  BUF  30.9   42    24  253   2    1     0    10       78   
110  Josh Allen  BUF  16.9   33    22  223   2    0     0     8       20   
116  Josh Allen  BUF  20.6   27    16  147   1    0     0    10       47   
122  Josh Allen  BUF  35.9   40    25  304   4    0     1    10       77   
129  Josh Allen  BUF  25.0   26    15  172   2    2     0     6       41   

     RUSHTD  WEEK  
68        0     8  
73        2     9  
82        0    10  
94        0    11  
96        1    12  
110       0    13  
116       1    14  
122       0    15  
129       1    16

In [11]:
df_ja_postbye.describe()

FPTS        ATT       COMP         YDS        TD  INT      300+  \
count   9.000000   9.000000   9.000000    9.000000  9.000000  9.0  9.000000   
mean   23.322222  33.000000  20.000000  227.666667  1.666667  1.0  0.222222   
std     7.167422   7.211103   5.291503   59.283219  1.118034  1.0  0.440959   
min    12.600000  25.000000  13.000000  147.000000  0.000000  0.0  0.000000   
25%    19.600000  27.000000  16.000000  197.000000  1.000000  0.0  0.000000   
50%    21.600000  33.000000  18.000000  218.000000  2.000000  1.0  0.000000   
75%    26.800000  40.000000  24.000000  253.000000  2.000000  2.0  0.000000   
max    35.900000  43.000000  29.000000  330.000000  4.000000  2.0  1.000000   

            RUSH    RUSHYDS    RUSHTD       WEEK  
count   9.000000   9.000000  9.000000   9.000000  
mean    7.555556  54.333333  0.555556  12.000000  
std     2.455153  28.801042  0.726483   2.738613  
min     3.000000   7.000000  0.000000   8.000000  
25%     6.000000  41.000000  0.000000  10.000000  
50%     8.000000  49.000000  0.000000  12.000000  
75%    10.000000  78.000000  1.000000  14.000000  
max    10.000000  86.000000  2.000000  16.000000

I will preface this by saying an average of 23.32 FPTS per game is still really good (for fantasy football). However since Allen was drafted so high, we're allowed to nitpick a little. The difference in Allen's performance was over 6 FPTS per game. But why? His passing stats took a hit, but his rushing stats were a bit better (not enough to make up the FPTS difference). Also, there's a good chance the coaching staff would preferred if he ran a little less as his aggressive running style subjugated him to big hits from opposing defenses.

Two notable stats that show his drop in performance are actually not listed in our stat table, but use the numbers above, are completion percentage (COMP/ATT) and TD-to-INT ratio (TD/INT).

Josh Allen's completion percentage dropped from 67.3 to 60.36 after Buffalo's bye week, which may not seem like much, until you check out the completion percentage of other QBs in the league (https://sports.yahoo.com/nfl/stats/individual/?sortStatId=COMPLETION_PERCENTAGE&selectedTable=0). If Josh Allen maintained 67.3 for the season, he would've ranked 5. 60.36 ranks 31st. Altogether, his rate of 63.3 was in the bottom half of the league. He also threw 17 TDs against 4 INTs before the bye. Thats more than 4 TDs for every INT. After the bye, that rate dropped to 1.8.

So Josh Allen was far less efficient passing the ball after the teams bye week. Was there a notable change in the offense? Well, Josh Allen did suffer an injury in his shoulder. While he did play through it, it wouldn't be surprising if it affected his passing a bit.

However that injury happened towards the end of Buffalo's week 9 game. If we were to divide the season after week 9, things look very different.

In [12]:
df_ja_preinj = df_ja.iloc[[0,1,2,3,4,5,6,7]]
df_ja_preinj.describe()

FPTS        ATT       COMP         YDS        TD      INT  \
count   8.000000   8.000000   8.000000    8.000000  8.000000  8.00000   
mean   27.925000  37.250000  23.875000  300.375000  2.375000  1.00000   
std     5.193884  11.411147   8.773947   84.373214  1.407886  0.92582   
min    19.600000  25.000000  13.000000  205.000000  0.000000  0.00000   
25%    25.925000  31.000000  18.750000  216.750000  1.750000  0.00000   
50%    26.750000  35.000000  23.000000  307.000000  2.500000  1.00000   
75%    30.650000  38.500000  26.250000  346.750000  3.250000  2.00000   
max    36.200000  63.000000  42.000000  424.000000  4.000000  2.00000   

           300+       RUSH    RUSHYDS    RUSHTD      WEEK  
count  8.000000   8.000000   8.000000  8.000000  8.000000  
mean   0.500000   7.750000  49.000000  0.500000  4.750000  
std    0.534522   3.615443  23.058931  0.755929  2.815772  
min    0.000000   1.000000  10.000000  0.000000  1.000000  
25%    0.000000   5.750000  39.500000  0.000000  2.750000  
50%    0.500000   8.500000  48.000000  0.000000  4.500000  
75%    1.000000  10.250000  59.500000  1.000000  6.500000  
max    1.000000  12.000000  86.000000  2.000000  9.000000

In [13]:
df_ja_postinj = df_ja.iloc[[8,9,10,11,12,13,14]]
df_ja_postinj.describe()

FPTS        ATT       COMP         YDS        TD       INT  \
count   7.000000   7.000000   7.000000    7.000000  7.000000  7.000000   
mean   23.357143  34.000000  21.285714  232.285714  1.857143  0.714286   
std     8.010588   7.571878   5.154748   67.527067  1.069045  0.951190   
min    12.600000  26.000000  15.000000  147.000000  1.000000  0.000000   
25%    18.750000  27.000000  17.000000  184.500000  1.000000  0.000000   
50%    21.600000  33.000000  22.000000  223.000000  2.000000  0.000000   
75%    27.950000  41.000000  24.500000  278.500000  2.000000  1.500000   
max    35.900000  43.000000  29.000000  330.000000  4.000000  2.000000   

           300+       RUSH    RUSHYDS    RUSHTD       WEEK  
count  7.000000   7.000000   7.000000  7.000000   7.000000  
mean   0.285714   7.571429  50.571429  0.428571  13.000000  
std    0.487950   2.699206  30.292228  0.534522   2.160247  
min    0.000000   3.000000   7.000000  0.000000  10.000000  
25%    0.000000   6.000000  30.500000  0.000000  11.500000  
50%    0.000000   8.000000  47.000000  0.000000  13.000000  
75%    0.500000  10.000000  77.500000  1.000000  14.500000  
max    1.000000  10.000000  84.000000  1.000000  16.000000

The difference pre injury and post injury is that Josh Allen threw less after hurting his shoulder. His completion percentage and TD-to-INT ratio look very similar. While there was an argument that this injury affected his accuracy, there was also an argument that Josh Allen will be too aggressive on occasion. That same aggressiveness is the reason Allen is as great as he is, so you have to take the bad with the good.

So we have two charts (pre-bye, post-bye) suggesting an efficient super Josh Allen is possible. We also have two charts (pre-injury, post-injury) that suggest a less than efficient but healthier Josh Allen may have been enough to grab QB1 for the third straight year. We need a tie breaker.

https://sports.yahoo.com/nfl/players/30977/

After this season, Josh Allen has played 5 seasons in the NFL. The first 2 were learning experiences. His third year (2020) is when he "broke out" and became a superstar. It was his best year throwing the football, with career highs in most categories, including efficiency. 69.2 completion rating was fourth best that season (would've been second best this season). He decidedly ran the ball a lot less. In his fourth season, his passing stats were down, probably due to defenses taking him a lot more seriously, but he supplemented that with career highs running the ball. This season (his fifth) was statistically similar to last season, in some ways even better. This is with a shoulder injury slowing him down.

So efficient Josh Allen is more of a fantasy. But aggressive (and healthy) Josh Allen is really good too. I don't see why he can't claim the QB1 title next year. Personally, I don't have a (recent) history of drafting QBs early, Josh Allen is a great choice for someone who doesn't want to have to guess who the next great QB might be. However, if he drops in the rankings and is available in a reasonable round, I'd have no trouble picking him (depending on Buffalo's offseason moves).

Buffalo has tried for years to build a good running game to complement. They'll probably try again this offseason. Perhaps a really good running back might take away from Allen, but probably not too much. They could also use another receiver to complement Steon Diggs. Diggs' arrival triggered Allen's break out year in 2020. Another high tier receiver can take this team to the next level. Gabe Davis was thought to be that guy, but this year proved otherwise.

### Lamar Jackson

In [14]:
Image(url="lj.jpg", width=948, height=632)

Drafting Lamar Jackson as a QB2 means chasing his ceiling. His ceiling was his 2019 MVP season where he threw for 3127 yards with 36 TDs against only 6 Int at a 66.1 completion rate, in addition to running for 1206 yards and 7 TDs. While those numbers draw Michael Vick comparisons, almost all of them beat Vick's career highs. All of this happened in only his second year. In fantasy, he was the QB1 on the year, scoring nearly 75 points more than the next guy. Fantasy tends to favor rushing QBs, but Lamar Jackson has taken great strides to improve his already underrated passing. He's aware of the threat of him running, and uses that to manipulate the defense. He would do well with a top receiver, which Baltimore believed they had with the emerging Rashod Bateman. They were so confident, they traded away Marquise Brown, a receiver Lamar Jackson had success with.

So how did 2022 turn out for Jackson? Well, not good. He was QB13 on the year (QB6 for per-game)

In [15]:
df_lj = df2[df2['TEAM']=='BAL']
df_lj = df_lj[['NAME','TEAM','FPTS','ATT','COMP','YDS','TD','INT','300+','RUSH','RUSHYDS','RUSHTD','WEEK']]
df_lj.describe()

FPTS        ATT       COMP         YDS         TD        INT  \
count  12.000000  12.000000  12.000000   12.000000  12.000000  12.000000   
mean   20.258333  27.166667  16.916667  186.833333   1.416667   0.583333   
std    11.461989   9.183318   6.501165   78.218614   1.311372   0.668558   
min     1.300000   4.000000   3.000000   11.000000   0.000000   0.000000   
25%    15.625000  27.250000  15.000000  141.250000   0.750000   0.000000   
50%    17.300000  29.500000  17.500000  209.500000   1.000000   0.500000   
75%    22.125000  32.000000  20.250000  223.000000   2.250000   1.000000   
max    42.600000  38.000000  27.000000  318.000000   4.000000   2.000000   

            300+       RUSH     RUSHYDS     RUSHTD       WEEK  
count  12.000000  12.000000   12.000000  12.000000  12.000000  
mean    0.083333   9.333333   63.666667   0.250000   6.750000  
std     0.288675   3.393398   34.240681   0.452267   3.957157  
min     0.000000   1.000000    9.000000   0.000000   1.000000  
25%     0.000000   8.500000   40.000000   0.000000   3.750000  
50%     0.000000  10.500000   66.000000   0.000000   6.500000  
75%     0.000000  11.000000   83.750000   0.250000   9.500000  
max     1.000000  14.000000  119.000000   1.000000  13.000000

In [16]:
df_lj

NAME TEAM  FPTS  ATT  COMP  YDS  TD  INT  300+  RUSH  RUSHYDS  \
5    Lamar Jackson  BAL  21.2   30    17  213   3    1     0     6       17   
11   Lamar Jackson  BAL  42.6   29    21  318   3    0     1     9      119   
21   Lamar Jackson  BAL  40.4   29    18  218   4    1     0    11      107   
38   Lamar Jackson  BAL  15.1   29    20  144   1    2     0    11       73   
45   Lamar Jackson  BAL  15.8   32    19  174   1    1     0    12       58   
51   Lamar Jackson  BAL  17.1   32    17  210   1    1     0     7       77   
63   Lamar Jackson  BAL  10.7   16     9  120   0    0     0    10       59   
67   Lamar Jackson  BAL  21.8   38    27  238   2    0     0     9       43   
77   Lamar Jackson  BAL  17.5   22    12  133   1    0     0    11       82   
93   Lamar Jackson  BAL  16.5   33    24  209   0    1     0    11       31   
100  Lamar Jackson  BAL  23.1   32    16  254   1    0     0    14       89   
112  Lamar Jackson  BAL   1.3    4     3   11   0    0     0     1        9   

     RUSHTD  WEEK  
5         0     1  
11        1     2  
21        1     3  
38        0     4  
45        0     5  
51        0     6  
63        0     7  
67        0     8  
77        0     9  
93        1    11  
100       0    12  
112       0    13

So the first three weeks were more than promising. Jackson scored at a perhaps unsustainable rate. But it all fell apart afterward. What happened after week three? The key to a great fantasy season for Lamar was always going to start with Bateman, but Bateman only plays in six games this season. He starts missing games week 5, and doesn't play much week 4 due to a foot injury.

It should be noted while Lamar struggled in fantasy, he did win 8 of his 12 games.

A healthy Jackson and Bateman could make for a good 2023 season in Baltimore. They've also hired a new offensive coordinator whose been known to be more aggressive throwing downfield. This could help elevate Jackson's passing game. Baltimore would also help themselves out by adding more receiver depth, so Jackson won't be scraping by to get a win.

### Justin Herbert

In [17]:
Image(url="jh.jpg")

It's not particularly common to see a QB put up top tier numbers in his first year, but Justin Herbert did that, earning him Rookie of the Year honors. He followed that up with a sophomore season where he threw for more than 5000 yards. A highly accurate passer with a rocket for an arm, he has enough tools to rival Patrick Mahomes.

So how was Justin Herbert's 2022 season?

In [18]:
df_jh = df2[df2['TEAM']=='LAC']
df_jh = df_jh[['NAME','TEAM','FPTS','ATT','COMP','YDS','TD','INT','300+','RUSH','RUSHYDS','RUSHTD','WEEK']]
df_jh.describe()

FPTS        ATT       COMP         YDS         TD        INT  \
count  16.000000  16.000000  16.000000   16.000000  16.000000  16.000000   
mean   16.968750  41.375000  28.250000  271.875000   1.437500   0.625000   
std     5.941685   8.655441   5.732946   53.438906   1.030776   0.619139   
min     6.500000  28.000000  21.000000  196.000000   0.000000   0.000000   
25%    13.075000  34.000000  23.750000  233.250000   1.000000   0.000000   
50%    17.400000  42.500000  27.500000  276.500000   1.000000   1.000000   
75%    20.425000  47.250000  33.000000  306.250000   2.000000   1.000000   
max    28.800000  57.000000  39.000000  367.000000   3.000000   2.000000   

            300+      RUSH    RUSHYDS  RUSHTD       WEEK  
count  16.000000  16.00000  16.000000    16.0  16.000000  
mean    0.250000   3.25000   9.250000     0.0   9.062500  
std     0.447214   1.65328  10.920928     0.0   5.208567  
min     0.000000   1.00000  -5.000000     0.0   1.000000  
25%     0.000000   1.75000   1.750000     0.0   4.750000  
50%     0.000000   4.00000   7.000000     0.0   9.500000  
75%     0.250000   5.00000  14.000000     0.0  13.250000  
max     1.000000   5.00000  38.000000     0.0  17.000000

In [19]:
df_jh

NAME TEAM  FPTS  ATT  COMP  YDS  TD  INT  300+  RUSH  RUSHYDS  \
3    Justin Herbert  LAC  23.3   34    26  279   3    0     0     4        1   
15   Justin Herbert  LAC  24.5   48    33  334   3    1     1     2        1   
27   Justin Herbert  LAC  13.6   45    25  297   1    1     0     1        7   
35   Justin Herbert  LAC  21.1   39    27  340   2    0     1     4       -5   
46   Justin Herbert  LAC  14.4   34    22  228   1    0     0     5       13   
56   Justin Herbert  LAC   9.4   57    37  238   0    1     0     2        9   
59   Justin Herbert  LAC  18.6   51    33  293   2    1     0     3       22   
79   Justin Herbert  LAC  13.1   43    30  245   1    1     0     1        3   
86   Justin Herbert  LAC  13.0   35    21  196   1    1     0     5       22   
92   Justin Herbert  LAC  20.2   30    23  280   2    1     0     5       17   
98   Justin Herbert  LAC  28.8   47    35  274   3    0     0     4       38   
108  Justin Herbert  LAC  18.1   47    28  335   1    0     1     5        7   
117  Justin Herbert  LAC  19.5   51    39  367   1    0     1     4        8   
126  Justin Herbert  LAC  10.7   42    28  197   0    2     0     1        2   
132  Justin Herbert  LAC   6.5   31    24  235   0    1     0     5        1   
135  Justin Herbert  LAC  16.7   28    21  212   2    0     0     1        2   

     RUSHTD  WEEK  
3         0     1  
15        0     2  
27        0     3  
35        0     4  
46        0     5  
56        0     6  
59        0     7  
79        0     9  
86        0    10  
92        0    11  
98        0    12  
108       0    13  
117       0    14  
126       0    15  
132       0    16  
135       0    17

Justin Herbert's season wasn't great. A lot of Herbert's problems mirrow Lamar Jackson's. Herbert's receivers had trouble staying healthy this season, and the options behind them were not great